### PR TITLE
Omar/setup

### PR DESCRIPTION
**Pull request by OmarG247**
- setup of the cool karts

### CODE REVIEW

File: v2/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import logo from './assets/trajectory_logo.svg';
+import buttonDown from './assets/button-down.svg';
+import GOTrainLight from './assets/gotrain-light.svg';
+import VIALight from './assets/via-light.svg';
+import StreetCarLight from './assets/streetcar-light.svg';
+import SubwayLight from './assets/subway-light.svg';
+import BusLight from './assets/bus-light.svg';
+import searchLight from './assets/search-light.svg';
+
+const transitImagesLight: any = {
```
Comment (dellisd) [inline v2/src/App.tsx:12]: Define an interface for this structure

File: v2/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  via: VIALight,
+  streetcar: StreetCarLight,
+  subway: SubwayLight,
+  bus: BusLight
+}
+
+const App = () => {
+  const [dropdown, activateDropdown] = useState(false);
+  const [search, activateSearch] = useState(false);
+  const [activeTransit, setActiveTransit]: any = useState({
```
Comment (dellisd) [inline v2/src/App.tsx:23]: Define another interface here

File: v2/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import logo from './assets/trajectory_logo.svg';
+import buttonDown from './assets/button-down.svg';
+import GOTrainLight from './assets/gotrain-light.svg';
+import VIALight from './assets/via-light.svg';
+import StreetCarLight from './assets/streetcar-light.svg';
+import SubwayLight from './assets/subway-light.svg';
+import BusLight from './assets/bus-light.svg';
+import searchLight from './assets/search-light.svg';
+
+const samepleVehicles = [
```
Comment (dellisd) [inline v2/src/App.tsx:13]: Is there an interface for this?

File: v2/src/Interfaces.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+export interface TransitIconImages {
+  [gotrain: string]: string,
```
Comment (dellisd) [inline v2/src/Interfaces.ts:2]:
```suggestion
  gotrain: string,
```

File: v2/src/Interfaces.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  via: string,
+  streetcar: string,
+  subway: string,
+  bus: string
+}
+
+export interface ActiveTransit {
+  [gotrain: string]: boolean,
```
Comment (dellisd) [inline v2/src/Interfaces.ts:10]:
```suggestion
  gotrain: boolean,
```

Was this the string indexing thing you were talking about?

File: v2/src/App.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      </div>
+      <div className="controls">
+        <div className="side-menu-container">
+          <div className="side-menu-header-container">
+            <h2 className="side-menu-header">
+              Active transit
+            </h2>
+            <img 
+              onClick={() => activateDropdown(!dropdown)} 
+              className={`side-menu-button ${dropdown ? 'dropdown-on' : 'dropdown-off'}`}
```
Comment (dellisd) [inline v2/src/App.tsx:83]: I highly recommend installing and using `clsx`

File: v2/src/Interfaces.ts (hunk 5 above)
Comment (OmarG247) [reply v2/src/Interfaces.ts:10]: yeah, if you don't flag the key as a string, it doesn't let you index with a string lmao

File: v2/src/App.tsx (hunk 3 above)
Comment (OmarG247) [reply v2/src/App.tsx:13]: This is just sample data to play with the carousels, don't think we need one

Comment (dellisd) [reply v2/src/App.tsx:14]: Once we get actual data we'll be able to reuse the interface. Would be useful to define it right away I think

File: v2/src/Interfaces.ts (hunk 5 above)
Comment (dellisd) [reply v2/src/Interfaces.ts:10]: Why do you need to index it with a string?

Comment (OmarG247) [reply v2/src/Interfaces.ts:10]: On the function generateIcon I use the icon object with all the URLs

File: v2/src/App.tsx (hunk 3 above)
Comment (OmarG247) [reply v2/src/App.tsx]: Done 👍

File: v2/src/App.tsx (hunk 6 above)
Comment (OmarG247) [reply v2/src/App.tsx]: Done on all cases 👍

**Pull request merged**